ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java

```diff
@@ -67,7 +66,18 @@ private void read(String suffix, int expectedTokenCount, Function<String[], Void
                     continue;
                 }
 
+                //skip first token
+
                 String[] tokens = trimedLine.split("( )+");
+
+                //check if it is the right network
+                if (tokens.length > 0) {
+                    int networkNum = Integer.parseInt(tokens[0]);
+                    if (variantIndex != networkNum) {
+                        continue;
+                    }
+                }
```

Comment:
Move this block of code after the next check: after this check you know that the number of token is stricly positive, so the code will be simpler:
```
                    int networkNum = Integer.parseInt(tokens[0]);
                    if (variantIndex != networkNum) {
                        continue;
                    }
```